tmail-backend/mailbox/encrypted/mailbox-encrypted-postgres/src/main/java/com/linagora/tmail/encrypted/postgres/PostgresEncryptedEmailStoreDAO.java

```diff
@@ -0,0 +1,106 @@
+package com.linagora.tmail.encrypted.postgres;
+
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.ENCRYPTED_ATTACHMENT_METADATA;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.ENCRYPTED_HTML;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.ENCRYPTED_PREVIEW;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.HAS_ATTACHMENT;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.MESSAGE_ID;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.POSITION_BLOB_ID_MAPPING;
+import static com.linagora.tmail.encrypted.postgres.table.PostgresEncryptedEmailStoreModule.PostgresEncryptedEmailStoreTable.TABLE_NAME;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.mailbox.postgres.PostgresMessageId;
+import org.jooq.Record;
+import org.jooq.impl.DSL;
+import org.jooq.postgres.extensions.types.Hstore;
+
+import com.google.common.collect.ImmutableMap;
+import com.linagora.tmail.encrypted.EncryptedAttachmentMetadata;
+import com.linagora.tmail.encrypted.EncryptedEmailContent;
+import com.linagora.tmail.encrypted.EncryptedEmailDetailedView;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import scala.compat.java8.OptionConverters;
+
+public class PostgresEncryptedEmailStoreDAO {
+    private static final String EMPTY_ENCRYPTED_ATTACHMENT_METADATA = null;
+
+    private final PostgresExecutor postgresExecutor;
+    private final BlobId.Factory blobIdFactory;
+
+    @Inject
```

Comment:
Tip: You can use `@Singleton` here
and don't need
```
bind(PostgresEncryptedEmailStoreDAO.class).in(Scopes.SINGLETON);
```

Reply:
Result is the same in the end though?

Reply:
yes, same result, single instance